src/core/request.js

```diff
@@ -44,13 +44,15 @@ Request.prototype.request = function(servers, callback_) {
   const callback = typeof servers === 'function' ? servers : callback_;
 
   this.emit('before');
+
+  const wasRequested = this.requested;
```

Comment:
Is there a reason we have to call `callback` even if this was already requested? Could we just throw an exception if request gets called twice?